dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/dataelement/DataElementDefaultDimensionPopulator.java

```diff
@@ -107,9 +110,28 @@ public void executeInTransaction() {
     for (DataElement dataElement : dataElements) {
       if (dataElement.getCategoryCombo() == null) {
         dataElement.setCategoryCombo(categoryCombo);
-
         dataElementService.updateDataElement(dataElement);
       }
     }
+
+    if (!hasCurrentUser) {
+      clearContext();
+    }
+  }
+
+  private static void injectUserInSecurityContext(SystemUser actingUser) {
+    Authentication authentication =
+        new UsernamePasswordAuthenticationToken(actingUser, "", actingUser.getAuthorities());
+    SecurityContext context = SecurityContextHolder.createEmptyContext();
+    context.setAuthentication(authentication);
+    SecurityContextHolder.setContext(context);
+  }
+
+  private static void clearContext() {
+    SecurityContext context = SecurityContextHolder.getContext();
+    if (context != null) {
+      SecurityContextHolder.getContext().setAuthentication(null);
+    }
+    SecurityContextHolder.clearContext();
```

Comment:
This feels like code that does exist elsewhere. I get we need this as `static` so if we have not yet created such helpers as static methods these should be put somewhere so we can reuse this code